addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -896,11 +897,38 @@ var FieldX2Many = AbstractField.extend({
         this.lastInitialEvent = undefined;
         if (Object.keys(changes).length) {
             this.lastInitialEvent = ev;
-            this._setValue({
+            var def = this._setValue({
                 operation: 'UPDATE',
                 id: ev.data.dataPointID,
                 data: changes,
             });
+
+            // Correctly resets the cursor
+            var self = this;
+            var cursor, fieldName;
+            var editableId =  this.renderer.getEditableRecordID();
+            var datapoint = _.find(this.renderer.state.data, {id: editableId});
```

Comment:
It seems to me this.value.data this is modified before and therefore it can not be used to retrieve the line being edited